myapp/src/services/adminService.js

```diff
@@ -1,104 +1,166 @@
+      throw {
+        status: 409,
+        message: `Report doesn't exists or is already accepted : ${id}`,
+      };
+    }
+    const reviewId = report[0]["review_id"];
+    try {
+      await db.query("BEGIN");
+      let result = await db.query({
+        text: `update report 
                     set status = $1, action = $2, admin_id = $3, respond_date = now()
                     where id = $4 returning *`,
-                values: [
-                    1,
-                    body.action,
-                    body.adminId,
-                    id,
-                ]
-            });
-            result = {
-                report: result.rows[0],
-            };
-            if(body.action === 'delete'){
-                const deleteResult = await reviewService.deleteReview(reviewId);
-                result.deleteRows = deleteResult;
-            }
-            await db.query('COMMIT');
-            return result;
-        }catch(err){
-            await db.query('ROLLBACK');
-            throw new Error(err);
-        }
-    },
+        values: [1, body.action, body.adminId, id],
+      });
+      result = {
+        report: result.rows[0],
+      };
+      if (body.action === "delete") {
+        const deleteResult = await reviewService.deleteReview(reviewId);
+        result.deleteRows = deleteResult;
+      }
+      await db.query("COMMIT");
+      return result;
+    } catch (err) {
+      await db.query("ROLLBACK");
+      throw new Error(err);
+    }
+  },
+  /** get facility registration request by id - possible for admin to view content */
+  getFacilityRegistrationRequest: async (id) => {
+    const query = {
+      text: `SELECT * FROM facility_registration_request WHERE id = $1`,
+      values: [id],
+    };
+    const result = await db.query(query);
+    return result.rows[0];
+  },
 
-}
\ No newline at end of file
+  /** get all facility registration requests */
+  getAllFacilityRegistrationRequests: async () => {
+    const query = `SELECT * FROM facility_registration_request ORDER BY created_at DESC`;
+    const result = await db.query(query);
+    return result.rows;
+  },
+  /** accept a facility registration request */
+  acceptFacilityRegistrationRequest: async (id, adminId) => {
+    const request = await module.exports.getFacilityRegistrationRequest(id);
+    if (!request || request.status !== 0) {
+      throw { status: 404, message: `Request doesn't exist or is not pending` };
+    }
+
+    const data = JSON.parse(request.content);
+    try {
+      await db.query("BEGIN");
+
+      // Create the facility and related entries
+      const facility = await facilityService.createFacility(data);
+
+      // Update the registration request status
+      const query = {
+        text: `UPDATE facility_registration_request SET status = $1, respond_date = NOW(), updated_at = NOW() WHERE id = $2 RETURNING *`,
+        values: [1, id],
+      };
+      const result = await db.query(query);
+
```

Comment:
Here, also insert the facility-manager relationship
`"insert into manages (user_id, facility_id) values ($1, $2)", [request.author_id, facility.id]`